src/main.cpp

```diff
@@ -173,8 +173,21 @@ int main(int argc, char **argv) {
     init_ui();
   }
 
-  if (NEEDROOT && (geteuid() != 0))
+#if NEEDROOT == 1
+  if (geteuid() != 0)
     forceExit(false, "You need to be root to run NetHogs!");
+#else
+  char exe_path[PATH_MAX];
+  unsigned int caps[5] = {0};
+
+  readlink("/proc/self/exe", exe_path, PATH_MAX);
+  getxattr(exe_path, "security.capability", (char *)caps, sizeof(caps));
+
+  if (((caps[1] >> CAP_NET_ADMIN) & 1) != 1)
+    forceExit(false, "You need to enable cap_net_admin (and cap_net_raw) to run NetHogs!");
```

Comment:
Nice! If we can further improve this message to also make sense to users who aren't aware of capabilities, I think we can remove the `NEEDROOT` switch entirely!